Rewrite "Member Registration and Login" for V14 (and without macros) (#6171)
umbraco/UmbracoDocs: 14/umbraco-cms/tutorials/members-registration-and-login.md

  login functionality for the frontend of your application.
---

# Member Registration and Login

Using tools available on a fresh install of Umbraco CMS, you can create a frontend-based registration and login functions and restrict access to specific areas of your site based on this system.

By the end of this tutorial, you will know how to implement a basic register/login functionality on your website, hide pages from non-logged-in members, and assign newly registered members to specific member groups.

## Install Umbraco and The Starter Kit

Make sure you have the latest [dotnet templates](../fundamentals/setup/install/install-umbraco-with-templates.md) installed.

Create an empty directory, open a terminal and run:

```bash
dotnet new umbraco
dotnet add package Umbraco.TheStarterKit
dotnet run
```

Complete the installer and login to the backoffice.

## Create partial views for Registration and Login

> Navigate to the Settings section in the backoffice. Locate the Partial Views entry under Templating in the left hand section, and click the "+" to create a new partial view.

Reviewer: ```suggestion
1. Navigate to the Settings section in the backoffice.
2. Locate the Partial Views entry under Templating in the left-hand section.
3. click the "+" to create a new partial view.
```

> In the dialog that follows, choose _New partial view from snippet..._:

Reviewer: ```suggestion
 4. Choose _New partial view from snippet..._: in the dialog that follows
```

![Create a new partial view](images/v14-create-partial-view-from-snippet.png)

> Then pick the _Login_ snippet in the next dialog:

Reviewer: ```suggestion
5. Pick the _Login_ snippet in the next dialog.
``.

![Create a new partial view from the Login snippet](images/v14-create-partial-view-from-login-snippet.png)

> Lastly, name the partial view "Login" and save it:

Reviewer: ```suggestion
6. Name the partial view "Login" and save it:
```

![Edit and save the "Login" partial view](images/v14-create-login-partial-view.png)

> Repeat these steps using the _Register Member_ and _Login Status_ snippets. Save these partial views as "Register" and "LoginStatus", respectively.

Reviewer: ```suggestion
Repeat the above steps using the _Register Member_ and _Login Status_ snippets. Save the partial views as "Register" and "LoginStatus", respectively.
```

## Render the partial views in the template

Locate the newly created "Login" template, and overwrite its content with this:

{% code title="Login.cshtml" %}
```cshtml
@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage
@{
    Layout = "master.cshtml";
}

@Html.Partial("~/Views/Partials/SectionHeader.cshtml")

<section class="section">

    <div class="container">

        <div class="col-md-3">
            <nav class="nav-bar nav-bar--list">
                @Html.Partial("~/Views/Partials/Navigation/SubNavigation.cshtml")
            </nav>
        </div>

        <div class="col-md-9">
            <article>
                @await Html.GetBlockGridHtmlAsync(Model, "bodyText")
            </article>
            <article>
                @await Html.PartialAsync("partials/loginStatus")
                @await Html.PartialAsync("partials/register")
                @await Html.PartialAsync("partials/login")
            </article>
        </div>
    </div>

</section>

<link rel="stylesheet" href="@Url.Content("~/css/umbraco-starterkit-blockgrid.css")" />
```
{% endcode %}

## Create the Register/Login page

Halfway there!

> Next, navigate to the Content section. Under the _Home_ node, create a new page based on the "Login" Document Type:

Reviewer: ```suggestion
1. Navigate to the Content section. 
2. Create a new page based on the "Login" Document Type Under the _Home_ node: 
```

![Creating the Register/Login page](images/v14-create-register-login-page.png)

> Save and publish the page. The Register and Login functionality is rendered by the "Login" template:

Reviewer: ```suggestion
3. Save and publish the page. 

The Register and Login functionality is rendered by the "Login" template:
```

  login functionality for the frontend of your application.
---

# Member Registration and Login

Using tools available on a fresh install of Umbraco CMS, you can create a frontend-based registration and login functions and restrict access to specific areas of your site based on this system.

By the end of this tutorial, you will know how to implement a basic register/login functionality on your website, hide pages from non-logged-in members, and assign newly registered members to specific member groups.

## Install Umbraco and The Starter Kit

Make sure you have the latest [dotnet templates](../fundamentals/setup/install/install-umbraco-with-templates.md) installed.

Create an empty directory, open a terminal and run:

```bash
dotnet new umbraco
dotnet add package Umbraco.TheStarterKit
dotnet run
```

Complete the installer and login to the backoffice.

## Create partial views for Registration and Login

1. Navigate to the Settings section in the backoffice.
2. Locate the Partial Views entry under Templating in the left-hand section.

> 3. click the "+" to create a new partial view.

Reviewer: ```suggestion
3. Click the "+" to create a new partial view.
```

* Check the current login status
* Log out a member

> We can take this a bit further and specify which parts of our website should be accessible to logged-in members. To do this, head on over to the Member section in the Backoffice, and create a new Member Group.

Reviewer: ```suggestion
We can take this a bit further and specify which parts of our website should be accessible to logged-in members. To do this:
 1. Go to the Member section in the Backoffice.
 2. Create a new Member Group.
```

![Create a new member group](images/v14-create-member-group.png)

> Give the group a name and save it:

Reviewer: ```suggestion
3. Give the group a name.
4. Save the Member Group.
```

![Naming the new member group](images/v14-create-member-group-step-2.png)

> Next, navigate back to the created Member. Assign the newly created Member Group and save the member:

Reviewer: ```suggestion
5. Navigate back to the created Member. 
6. Assign the newly created Member Group.
7. Save the member:
```

* Check the current login status
* Log out a member

We can take this a bit further and specify which parts of our website should be accessible to logged-in members. To do this:
 1. Go to the Member section in the Backoffice.
 2. Create a new Member Group.

![Create a new member group](images/v14-create-member-group.png)

3. Give the group a name.
4. Save the Member Group.

![Naming the new member group](images/v14-create-member-group-step-2.png)

5. Navigate back to the created Member. 
6. Assign the newly created Member Group.
7. Save the member:

![Assign the new Member group to the created Member](images/v14-assign-member-group.png)

Almost there!

> Now navigate to the Content section. Create a new page that should only be visible to "Premium" members.

Reviewer: ```suggestion
1. Navigate to the Content section. 
2. Create a new page that should only be visible to "Premium" members.
```

> Once created, click the menu icon (•••) to bring up the page options, and pick "Restrict Public Access":

Reviewer: ```suggestion
 3. Click the menu icon (•••) to bring up the page options, and pick "Restrict Public Access" once created.
```
